assets/js/atomic/blocks/product-elements/summary/edit.tsx

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { useBlockProps } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import Block from './block';
+import withProductSelector from '../shared/with-product-selector';
+import {
+	BLOCK_TITLE as label,
+	BLOCK_ICON as icon,
+	BLOCK_DESCRIPTION as description,
+} from './constants';
+import './editor.scss';
+import type { BlockAttributes } from './types';
+
+interface Props {
+	attributes: BlockAttributes;
+}
+
+const Edit = ( { attributes }: Props ): JSX.Element => {
+	const blockProps = useBlockProps();
+	return (
+		<div { ...blockProps }>
+			<Block { ...attributes } />
+		</div>
+	);
+};
+
+// @todo: Refactor this to remove the HOC 'withProductSelector()' component.
```

Comment:
I think we should provide more context for this to do explaining why we should remove the HOC.

Reply:
This todo applies to other product elements as well.

Reply:
Good call. I added more context, mentioned that other `product-elements` components might be affected as well and added a link to your initial suggestion above.